include/core/CMemoryDef.h

```diff
@@ -207,7 +207,9 @@ template<typename K, typename V, typename H, typename P, typename A>
 std::size_t dynamicSize(const boost::unordered_map<K, V, H, P, A>& t) {
     return memory::elementDynamicSize(t) +
            (t.bucket_count() * memory::storageNodeOverhead(t)) +
-           (t.size() * (sizeof(K) + sizeof(V) + memory::storageNodeOverhead(t)));
+           (t.size() * (sizeof(K) + sizeof(V) + memory::storageNodeOverhead(t))) +
+           (static_cast<std::size_t>(std::ceil(t.bucket_count() / 2.0 / sizeof(std::size_t))) *
+            sizeof(std::size_t));
```

Comment:
You can do this without resorting to floating point functions like this:

```
((t.bucket_count() + 2 * sizeof(std::size_t) - 1) / 2 * sizeof(std::size_t)) * sizeof(std::size_t)
```

The compiler can simplify that better at compile time and avoid a function call at runtime.

Also, this formula could go in a new function `std::size_t bucketGroupOverhead(std::size_t bucketCount)` that could then have a Doxygen comment explaining what it's doing and linking to the Boost unordered documentation. That function should get inlined by the compiler if it's defined higher up the same file, so won't really add any overhead.